### PR TITLE
Add fast path for jnp.repeat when repeats is a scalar constant.

### PR DESCRIPTION
**Pull request by petebu**
The default code-path generates a complicated soup of ops for this very simple case what can be implemented using reshapes and broadcasts only.

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -2883,6 +2883,16 @@ def repeat(a, repeats, axis=None, *, total_repeat_length=None):
       "To fix this, either specify a static value for `repeats`, or pass a static "
       "value to `total_repeat_length`.")
     repeats = np.ravel(repeats)
+
+    # Special case when repeats is a scalar.
+    if ndim(repeats) == 0:
+      input_shape = a.shape
+      a = reshape(a, np.insert(input_shape, axis, 1))
```
Comment (hawkinsp) [inline jax/_src/numpy/lax_numpy.py:2890]: `lax.expand_dims`

?

**Review by hawkinsp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (mattjj) [reply jax/_src/numpy/lax_numpy.py:2890]: +1

**Review by mattjj — APPROVED** [review]
Thanks for this improvement!

+1 to hawkinsp@'s comment.

Consider this optional, but since all untested code is (eventually) broken, I wonder how we might add a test for this. Maybe we could just call `make_jaxpr` and check the number of eqns is only ~3?

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (petebu) [reply jax/_src/numpy/lax_numpy.py]: That's better. Thanks!

**Comment by petebu** [comment]
> Thanks for this improvement!
> 
> +1 to hawkinsp@'s comment.
> 
> Consider this optional, but since all untested code is (eventually) broken, I wonder how we might add a test for this. Maybe we could just call `make_jaxpr` and check the number of eqns is only ~3?

Good idea. In fact, the original code was wrong but wasn't being called so the tests passed anyway. I've just committed a fix and added a test for the number of eqns.

**Comment by petebu** [comment]
Before this PR, `jnp.repeat(jnp.array([1,2,3,4]), repeats=2)` generated 21 Jaxpr eqns. Now it's down to 6.

**Review by hawkinsp — APPROVED** [review]

**Review by mattjj — APPROVED** [review]
Extra LGTM for an extra good PR.

**Pull request merged**